--- a/CPUMeter.c
+++ b/CPUMeter.c
@@ -131,15 +131,19 @@
       len = xSnprintf(buffer, sizeof(buffer), "%5.1f%% ", this->values[CPU_METER_KERNEL]);
       RichString_appendAscii(out, CRT_colors[METER_TEXT], "sy:");
       RichString_appendnAscii(out, CRT_colors[CPU_SYSTEM], buffer, len);
-      len = xSnprintf(buffer, sizeof(buffer), "%5.1f%% ", this->values[CPU_METER_NICE]);
-      RichString_appendAscii(out, CRT_colors[METER_TEXT], "ni:");
-      RichString_appendnAscii(out, CRT_colors[CPU_NICE_TEXT], buffer, len);
-      len = xSnprintf(buffer, sizeof(buffer), "%5.1f%% ", this->values[CPU_METER_IRQ]);
-      RichString_appendAscii(out, CRT_colors[METER_TEXT], "hi:");
-      RichString_appendnAscii(out, CRT_colors[CPU_IRQ], buffer, len);
+ 
       len = xSnprintf(buffer, sizeof(buffer), "%5.1f%% ", this->values[CPU_METER_SOFTIRQ]);
       RichString_appendAscii(out, CRT_colors[METER_TEXT], "si:");
       RichString_appendnAscii(out, CRT_colors[CPU_SOFTIRQ], buffer, len);
+
+      len = xSnprintf(buffer, sizeof(buffer), "%5.1f%% ", this->values[CPU_METER_IRQ]);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], "hi:");
+      RichString_appendnAscii(out, CRT_colors[CPU_IRQ], buffer, len);
+
+      len = xSnprintf(buffer, sizeof(buffer), "%5.1f%% ", this->values[CPU_METER_NICE]);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], "ni:");
+      RichString_appendnAscii(out, CRT_colors[CPU_NICE_TEXT], buffer, len);
+
       if (!isnan(this->values[CPU_METER_STEAL])) {
          len = xSnprintf(buffer, sizeof(buffer), "%5.1f%% ", this->values[CPU_METER_STEAL]);
          RichString_appendAscii(out, CRT_colors[METER_TEXT], "st:");
